### ISSUE TITLE
Pokémon Red/Blue trade/multiplayer not working 

### ISSUE THREAD

**Issue by Essnem1** [mgba-emu/mgba#2390]
Hey! So I’m very new to PC’s and emulators in general so sorry if this is rather basic. 

I am playing Pokemon red on mGBA. I was trying to trade between Red and Blue via the multiplayer window and I meet these two issues. 

1; when I try to enter the cable corner I received the default text telling me ABOUT the functionality. I’m assuming it doesn’t recognise the ‘cable’ and therefore it won’t allow the trade function. 

2; when I try to open a save file in the multiplayer window (2 of 2) it doesn’t recognise .ss0 files and only recognises .sav files. If I open a primary mGBA instead of a multiplayer window then it can find the .ss0 files. I can’t change the .ss0 files into .sav files either because when I do it regressed to a former save state. 

I hope these issues make sense.

**Comment by Essnem1** [comment]
![Uploading EBDFA92D-B53D-4822-B622-D2E89E7CE136.jpeg…]()

Issue 3. If I load a generic .sav and try to enter the cable club it stalls out like this and freezes

**Comment by Essnem1** [comment]
Please help I really want to do some trade evolutions that iv never been able to do on GBC lol

**Comment by Daniel-the-Spaniel** [comment]
I am also having a similar problem!!! I set up two multiplayer windows of Yellow Version to evolve certain Pokemon and the moment they link it stops responding to any keyboard input. The music still plays, and I could control both copies fine until they linked.

**Comment by CodeWithMa** [comment]
Same for trading with blue and yellow. Its stuck at this:

![image](https://user-images.githubusercontent.com/101021254/171637107-d1764c9a-1239-4f18-acca-d529fbd2799c.png)

**Comment by IdiotSandwichTheThird** [comment]
Yeah, I was really hoping this was something that'd work on an emulator in 2022. Oh well. Maybe in 2030.

**Comment by endrift** [comment]
@CodeWithMa that bug is #2290. This specific issue appears to be misunderstanding how the functionality works. @Essnem1, try downloading a dev build. It might be clearer what you're supposed to do on it.

**Comment by Dullstar** [comment]
I get the same result as @CodeWithMa when linking games in the exact same way that you would for GBA games. It seems to be that the stable versions have partial support in the sense that the games recognize that they're linked, but something goes wrong when you try to do anything with the linkage (if they're not linked at all, the receptionist in the Pokemon Center will simply say the area is reserved for 2 friends linked together by cable).

Out of curiosity, I tried a dev build and it seems to work, though (using 0.10-7542-b43509e70, which I'd already downloaded to confirm an unrelated issue I was having with 0.9.3 hadn't yet been fixed) I had to manually load the save for the Player 2 window, as it attempted to create a new one with a .sa2 extension instead of loading the existing .sav, which doesn't seem to happen when loading the game in the Player 1 window. It did not matter which game was loaded as Player 2; whichever one was Player 2 would need its save manually loaded (and then of course all those .sa2 files had to be cleaned up since I didn't actually want them). Didn't test latest dev to see if that .sa2 issue is fixed, but if you want I can do it later and open a new issue if it's still happening (I looked very briefly but didn't see one for it). That said, once the correct save was loaded it worked fine (tested between Red and Blue version).

**Comment by endrift** [comment]
> I get the same result as @CodeWithMa when linking games in the exact same way that you would for GBA games. It seems to be that the stable versions have partial support in the sense that the games recognize that they're linked, but something goes wrong when you try to do anything with the linkage (if they're not linked at all, the receptionist in the Pokemon Center will simply say the area is reserved for 2 friends linked together by cable).

That bug is #2290, not this one, as I said in my reply last night.

> Out of curiosity, I tried a dev build and it seems to work, though (using 0.10-7542-b43509e70, which I'd already downloaded to confirm an unrelated issue I was having with 0.9.3 hadn't yet been fixed) I had to manually load the save for the Player 2 window, as it attempted to create a new one with a .sa2 extension instead of loading the existing .sav, which doesn't seem to happen when loading the game in the Player 1 window. It did not matter which game was loaded as Player 2; whichever one was Player 2 would need its save manually loaded (and then of course all those .sa2 files had to be cleaned up since I didn't actually want them). Didn't test latest dev to see if that .sa2 issue is fixed, but if you want I can do it later and open a new issue if it's still happening (I looked very briefly but didn't see one for it). That said, once the correct save was loaded it worked fine (tested between Red and Blue version).

That's not a bug. That's it attempting to load the player 2 save so that you can load more than one save with the same ROM file. But that said, it is *confusing* and something that needs to be improved. It's also still not related to this issue, which I think is a misunderstanding and not an actual bug.
